node/pkg/common/batch_message.go

```diff
@@ -0,0 +1,107 @@
+package common
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type BatchMessageID struct {
+	EmitterChain  vaa.ChainID
+	TransactionID common.Hash
+}
+
+type BatchMessage struct {
+	BatchMessageID
+	Messages []*MessagePublication
+}
+
+func (b BatchMessageID) String() string {
+	return fmt.Sprintf("%d/%s",
+		b.EmitterChain,
+		hex.EncodeToString(b.TransactionID.Bytes()))
+}
+
+// 35 bytes for EmitterChain + TransactionID, 88 for MessagePublication
+const minBatchMsgLength = 123
+
+func (batch *BatchMessage) Marshal() ([]byte, error) {
+	buf := new(bytes.Buffer)
+
+	vaa.MustWrite(buf, binary.BigEndian, batch.EmitterChain)
+
+	buf.Write(batch.TransactionID[:])
+
+	vaa.MustWrite(buf, binary.BigEndian, uint8(len(batch.Messages)))
+
+	for _, msg := range batch.Messages {
+		msgBytes, err := msg.Marshal()
+		if err != nil {
+			return nil, err
+		}
+		lenBytes := len(msgBytes)
+
+		vaa.MustWrite(buf, binary.BigEndian, uint8(lenBytes))
```

Comment:
Good eye.

I was able to refactor this function out (to use the `vaa/structs.go` `Marshal()` rather than this similar function). I'm including checks for overflowing there.

Reply:
Added a check for this overflow to the `Unmarshal` code path of BatchVAAs in `structs.go` (I got rid of this file while refactoring, but this was a good idea, so I put it in the relevant place)